lib/feature/detail_card/detail_card_bloc.dart

```diff
@@ -472,24 +480,61 @@ class DetailCardBloc extends BlocBase {
     appBarEnumSubject.value = DetailCardAppBarEnum.comment;
   }
 
-  void sendComment() {
+Future<void> sendComment() async {
     if (!isChatCommentSubject.value) {
       return;
     }
-    final comment = Fragment$CommentFragment(
-      id: '1',
-      createdAt: '',
-      user: Fragment$UserFragment(
-        uid: '1',
-        fullName: 'Đinh Viết Khang',
-        createdAt: DateTime.now().toIso8601String(),
+    isLoadingAddCommentSubject.value = true;
+    await graphqlService.client.mutate$CreateComment(
+      Options$Mutation$CreateComment(
+        variables: Variables$Mutation$CreateComment(
+          idCard: cardSubject.value!.id,
+          content: commentController.text,
+        ),
+      ),
+    );
+    final resultGetCard = await graphqlService.client.mutate$GetCard(
+      Options$Mutation$GetCard(
+        variables: Variables$Mutation$GetCard(
+          idCard: idCard,
+        ),
+      ),
+    );
+    commentController.clear();
+    cardSubject.value = resultGetCard.parsedData?.getCard;
+    listCommentFragmentsSubject.value = [...cardSubject.value?.comments ?? []];
+    isLoadingAddCommentSubject.value = false;
+    scrollToListComment();
+  }
+
+  Future<void> deleteComment(String idComment) async {
+    isLoadingSubject.value = true;
+    await graphqlService.client.mutate$DeleteComment(
+      Options$Mutation$DeleteComment(
+        variables: Variables$Mutation$DeleteComment(
+          idComment: idComment,
+        ),
+      ),
+    );
+    final resultGetCard = await graphqlService.client.mutate$GetCard(
+      Options$Mutation$GetCard(
+        variables: Variables$Mutation$GetCard(
+          idCard: idCard,
+        ),
       ),
-      comment: commentController.text,
     );
-    listCommentFragmentsSubject.value = [
-      ...[comment],
-      ...listCommentFragmentsSubject.value,
-    ];
+    cardSubject.value = resultGetCard.parsedData?.getCard;
+    listCommentFragmentsSubject.value = [...cardSubject.value?.comments ?? []];
+    isLoadingSubject.value = false;
+    toastService.showText(message: 'Xóa thành công');
+  }
+
+  void scrollToListComment() {
+    Scrollable.ensureVisible(
+      dataKey.currentContext!,
```

Comment:
```suggestion
      dataKey.currentContext!,
```
đặt tên cái key này lại cho đúng ngữ cảnh của nó